src/amo/utils.js

```diff
@@ -50,7 +53,23 @@ export function fetchLandingAddons({ addonType, api, dispatch }) {
 
 export function loadLandingAddons({ store: { dispatch, getState }, params }) {
   const state = getState();
-  const addonType = apiAddonType(params.visibleAddonType);
+  try {
+    const addonType = apiAddonType(params.visibleAddonType);
+
+    return fetchLandingAddons({ addonType, api: state.api, dispatch });
+  } catch (err) {
+    return Promise.reject(err);
+  }
+}
 
-  return fetchLandingAddons({ addonType, api: state.api, dispatch });
+export function getErrorComponent(status) {
+  switch (status) {
+    case 401:
+      return NotAuthorized;
+    case 404:
+      return NotFound;
+    case 500:
```

Comment:
Again, not needed but I like to see the explicit mapping so I thought this was okay; if it's annoying or you'd like it removed I'm fine with that, though.